skbio/sequence/_sequence.py

```diff
@@ -491,7 +491,23 @@ def _string(self):
 
     def __init__(self, sequence, metadata=None,
                  positional_metadata=None):
-        if isinstance(sequence, Sequence):
+
+        if type(sequence) == np.ndarray or isinstance(sequence, np.ndarray):
```

Comment:
This check seems redundant. Did it speed things up when you benchmarked? I'd be surprised because `isinstance` is slow in the negative case, but you're optimizing for the positive case, which should be relatively quick via `isinstance` already.